File: ios/JSBridge/Sources/JSBridge/Commands/bottomnavigation/BottomNavigationService.swift
import SwiftUI

/// Snapshot of bottom navigation bar state, observed by the host app's SwiftUI views.
public struct BottomNavigationConfig: Sendable {
    public var isVisible: Bool = true

    public init(isVisible: Bool = true) {
        self.isVisible = isVisible
    }
}

/// Holds bottom bar visibility as an `ObservableObject` for SwiftUI integration.
///
/// Singleton because bottom bar visibility is app-wide shared state -- the same
/// bar is shown or hidden regardless of which WebView issues the command.
@MainActor
public final class BottomNavigationService: ObservableObject {
    @Published public var config = BottomNavigationConfig()

    public static let shared = BottomNavigationService()

    private init() {}

    public func configure(with config: BottomNavigationConfig) {
        self.config = config
    }

    public func setVisible(_ isVisible: Bool) {
        withAnimation(.easeInOut(duration: 0.3)) {
            self.config.isVisible = isVisible
        }
    }
}

File: android/jsbridge/src/main/java/net/kibotu/jsbridge/commands/bottomnavigation/BottomNavigationService.kt
package net.kibotu.jsbridge.commands.bottomnavigation

import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.SharedFlow


/**
 * Holds bottom navigation visibility as a [StateFlow].
 *
 * Singleton because visibility is app-wide shared state -- the same bar is
 * shown/hidden regardless of which WebView issues the command. The host
 * Activity collects [isVisible] to drive the bottom bar's actual visibility.
 */
object BottomNavigationService {

    val isVisible: SharedFlow<Boolean>
        field = MutableStateFlow<Boolean>(true)

    fun setVisible(visible: Boolean) {
        isVisible.value = visible
    }

    fun currentVisibility(): Boolean = isVisible.value
}